DatadogSessionReplay/Sources/Recorder/RecordingCoordinator.swift

```diff
@@ -93,14 +95,17 @@ internal class RecordingCoordinator {
     }
 
     private func evaluateRecordingConditions() {
-        if recordingEnabled && isSampled && currentRUMContext != nil {
+        if !isRecording && recordingEnabled && isSampled && currentRUMContext != nil {
+            isRecording = true
             recordingTrigger.startWatchingTriggers { [weak self] in
                 self?.didTrigger()
             }
-        } else {
+            updateHasReplay()
+        } else if isRecording {
+            isRecording = false
             recordingTrigger.stopWatchingTriggers()
+            updateHasReplay()
```

Comment:
So this change is to avoid updating the context through `updateHasReplay`, correct? 

What happens if we are recording and we receive a context update? It looks like since `isRecording` is `true`, it will stop the recording.

Reply:
Yeah - it got caught by the integration test. Great catch!

It should be fixed now.

Reply:
/suggestion The statements are difficult to understand and redundant with the `updateHasReplay()`. We could make it both easier to read and more straightforward with:
```swift
let hasReplay = isSampled && recordingEnabled
if !isRecording && hasReplay && currentRUMContext != nil {
    isRecording = true
    recordingTrigger.startWatchingTriggers { [weak self] in
        self?.didTrigger()
    }
    srContextPublisher.setHasReplay(hasReplay)
} else if isRecording && !hasReplay {
    isRecording = false
    recordingTrigger.stopWatchingTriggers()
    srContextPublisher.setHasReplay(hasReplay)
}

```
Also, I don't see the reason why adding `currentRUMContext` here?